### PR TITLE
Upgrade syntax to Python 3.7

### PR DESCRIPTION
**Pull request by nsoranzo**
using `ack --type=python -f | grep -v '^tools/' | xargs pyupgrade --py37-plus`

Also more refactoring of Python files in `packages/` .

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by bgruening** [comment]
Cool, but I guess we should wait until @jmchilton work has landed?

**Comment by nsoranzo** [comment]
Which one? This is very small compared to the black PR.

**Comment by jmchilton** [comment]
I have reservations about this but my open PRs are a non-issue here, feel free to merge.

**Comment by ic4f** [comment]
Can we, please, not merge this before #13108? I will have those conflicts resolved today, so it, hopefully, can be merged. The conflicts I'm currently resolving appear to be non-trivial: both branches have the same code that has been moved + refactored early on, followed by commits building on top of those changes. Maybe this PR won't add to it, but maybe it will.

File: lib/galaxy/datatypes/blast.py
Comment (bgruening) [inline lib/galaxy/datatypes/blast.py:105]: Puh, this looks not good.

File: lib/tool_shed/dependencies/tool/tag_attribute_handler.py
Comment (bgruening) [inline lib/tool_shed/dependencies/tool/tag_attribute_handler.py]: why not f-strings here?

Comment (bgruening) [inline lib/tool_shed/dependencies/tool/tag_attribute_handler.py]: why not f-strings here?

File: lib/tool_shed/galaxy_install/dependency_display.py
Comment (bgruening) [inline lib/tool_shed/galaxy_install/dependency_display.py]: or here

Comment (bgruening) [inline lib/tool_shed/galaxy_install/dependency_display.py]: or here

File: lib/galaxy/datatypes/blast.py
Comment (nsoranzo) [reply lib/galaxy/datatypes/blast.py:105]: Sorry, why?

**Comment by nsoranzo** [comment]
I've converted (almost) all format strings to f-strings. Ready for a re-review.

File: lib/galaxy/datatypes/blast.py
Comment (jdavcs) [reply lib/galaxy/datatypes/blast.py:105]:
```suggestion
            raise ValueError(f"Given no BLAST XML files, {repr(split_files)}, to merge into {output_file}")
```

https://peps.python.org/pep-0498/#s-r-and-a-are-redundant 
Just a suggestion - no strong opinion.

Comment (nsoranzo) [reply lib/galaxy/datatypes/blast.py:105]: No strong opinion either, but a quick test with `timeit` shows that using `repr()` is slower.

Comment (jdavcs) [reply lib/galaxy/datatypes/blast.py:105]: Let's keep the `!r` syntax then. Thanks for running the test!

File: packages/meta/setup.py
Comment (jdavcs) [inline packages/meta/setup.py:32]: All of these changes (see other packages too) don't look right. `get_var` references `init_contents` that may not exist. I don't think these should be moved out of the context manager.

Comment (nsoranzo) [reply packages/meta/setup.py:32]: `init_contents` does exist when `get_var()` is called, you can add a `breakpoint()` at line 27 if you don't believe me. Also, this is tested by the `test_galaxy_packages` CircleCI tests.

Comment (nsoranzo) [reply packages/meta/setup.py:32]: Also, my near future plan is to replace as much as possible of the `setup.py` file in packages with `setup.cfg`, so these will hopefully go away soonish.

Comment (jdavcs) [reply packages/meta/setup.py:32]: > `init_contents` does exist when `get_var()` is called, you can add a `breakpoint()` at line 27 if you don't believe me. Also, this is tested by the `test_galaxy_packages` CircleCI tests.

I didn't mean variable scoping; what I meant was that if the file is not read, the variable won't exist. But then, in that case an error will be raised before reaching the reference, so it doesn't matter. I think it seemed to me that the indent made it a little more logical, but on second thought, maybe not. So, looks fine.

**Review by jdavcs — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks for the reviews @bgruening @ic4f !

**Pull request merged**
